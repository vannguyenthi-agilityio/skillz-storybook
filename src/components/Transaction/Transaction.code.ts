export const asDefault = `
import Transaction from '.';

/**
 * The used to render number an interface transaction.
 *
 */

 <Transaction
  label='Enrolled'
  content='Lorem ipsum dolor sit amet, consectetu'
  score={-3}
  timeDate='02:15 pm. February 12, 2023'
  icon={<FacebookIcon />}
  bgColorIcon='yellow.500'
 />
`;

export const transactionSlack = `
import Transaction from '.';
import { SlackIcon } from '@components';

/**
 * The used to render a transaction with Slack.
 *
 */

<Transaction
  label='Enrolled'
  content='Lorem ipsum dolor sit amet, consectetu'
  score={-3}
  timeDate='02:15 pm. February 12, 2023'
  icon={<SlackIcon />}
  bgColorIcon='green.400'
/>
`;

export const transactionJavascript = `
import Transaction from '.';
import { JavascriptIcon } from '@components';

/**
 * The used to render a transaction Javascript.
 *
 */

<Transaction
  label='Enrolled'
  content='Lorem ipsum dolor sit amet, consectetu'
  score={-3}
  timeDate='02:15 pm. February 12, 2023'
  icon={<JavascriptIcon />}
  bgColorIcon='purple.450'
/>
`;

export const transactionRuby = `
import Transaction from '.';
import { RubyIcon } from '@components';

/**
 * The used to render a transaction Ruby.
 *
 */

<Transaction
  label='Enrolled'
  content='Lorem ipsum dolor sit amet, consectetu'
  score={-3}
  timeDate='02:15 pm. February 12, 2023'
  icon={<RubyIcon />}
  bgColorIcon='orange.500'
/>
`;

export const transactionGoogle = `
import Transaction from '.';
import { GoogleIcon } from '@components';

/**
 * The used to render a transaction Google.
 *
 */

<Transaction
  label='Enrolled'
  content='Lorem ipsum dolor sit amet, consectetu'
  score={-3}
  timeDate='02:15 pm. February 12, 2023'
  icon={<GoogleIcon />}
  bgColorIcon='cyan.500'
/>
`;
