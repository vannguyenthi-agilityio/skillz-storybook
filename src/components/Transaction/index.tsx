// Components
import { Flex, Text, Divider } from '@chakra-ui/react';
import { StarLayoutIcon } from '@components';

// Interface
import type { ITransaction } from '@interfaces/transaction.interface';

// Helpers
import { convertPxToRem } from '@helpers/common';

export const Transaction = ({
  label = '',
  content = '',
  score,
  timeDate = '',
  bgColorIcon = '',
  icon,
  ...rest
}: ITransaction) => {
  return (
    <Flex
      maxW={convertPxToRem(900)}
      w='full'
      data-testid={`transaction`}
      flexDirection='column'
      {...rest}
    >
      <Flex
        alignItems='center'
        w='full'
        justifyContent='space-between'
        flexDirection={{ base: 'column', md: 'row' }}
      >
        <Flex
          justifyContent='center'
          alignItems='center'
        >
          <Flex
            bg={bgColorIcon}
            borderRadius='full'
            w={convertPxToRem(48)}
            h={convertPxToRem(48)}
            justifyContent='center'
            alignItems='center'
          >
            {icon}
          </Flex>
          <Flex
            ml={{ base: convertPxToRem(10), sm: convertPxToRem(20), md: convertPxToRem(26) }}
            flexDirection='column'
            color='grey.850'
            fontSize='base'
          >
            <Text fontWeight='light'>{label}:</Text>
            <Text fontWeight='bold'>{content}</Text>
          </Flex>
        </Flex>
        <Flex
          justifyContent='center'
          alignItems='center'
          fontWeight='normal'
        >
          <Text
            fontSize='sm'
            pl='3'
          >
            {timeDate}
          </Text>
          <Flex
            justifyContent='center'
            alignItems='center'
            ml={{ base: convertPxToRem(10), sm: convertPxToRem(20), md: convertPxToRem(50), '2xl': convertPxToRem(60) }}
          >
            <Text px='2'>{score}</Text>
            <StarLayoutIcon />
          </Flex>
        </Flex>
      </Flex>
      <Divider
        mt='5'
        borderColor='grey.150'
      />
    </Flex>
  );
};

export default Transaction;
