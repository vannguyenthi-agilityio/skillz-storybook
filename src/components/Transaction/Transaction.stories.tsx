import Storybook from '@storybook/react';

// Components
import Transaction from '.';
import { FacebookIcon, SlackIcon, JavascriptIcon, GoogleIcon, RubyIcon } from '@components';

// Code
import {
  asDefault,
  transactionSlack,
  transactionJavascript,
  transactionRuby,
  transactionGoogle
} from './Transaction.code';

export default {
  title: 'Components/Transaction',
  component: Transaction,
  parameters: {
    componentSource: {
      code: asDefault,
      language: 'typescript'
    }
  }
} as Storybook.ComponentMeta<typeof Transaction>;

const Template: Storybook.ComponentStory<typeof Transaction> = (args) => {
  return <Transaction {...args} />;
};

export const TransactionDefault = Template.bind({});
TransactionDefault.args = {
  label: 'Enrolled',
  content: 'Lorem ipsum dolor sit amet, consectetu',
  score: -3,
  timeDate: '02:15 pm. February 12, 2023',
  bgColorIcon: 'yellow.500',
  icon: <FacebookIcon />
};

export const TransactionSlack = Template.bind({});
TransactionSlack.args = {
  label: 'Enrolled',
  content: 'Lorem ipsum dolor sit amet, consectetu',
  score: -3,
  timeDate: '02:15 pm. February 12, 2023',
  bgColorIcon: 'green.400',
  icon: (
    <SlackIcon
      w='35'
      h='35'
    />
  )
};
TransactionSlack.parameters = {
  componentSource: {
    code: transactionSlack
  }
};

export const TransactionJavascript = Template.bind({});
TransactionJavascript.args = {
  label: 'Enrolled',
  content: 'Lorem ipsum dolor sit amet, consectetu',
  score: -3,
  timeDate: '02:15 pm. February 12, 2023',
  icon: <JavascriptIcon />,
  bgColorIcon: 'purple.450'
};
TransactionJavascript.parameters = {
  componentSource: {
    code: transactionJavascript
  }
};

export const TransactionRuby = Template.bind({});
TransactionRuby.args = {
  label: 'Enrolled',
  content: 'Lorem ipsum dolor sit amet, consectetu',
  score: -3,
  timeDate: '02:15 pm. February 12, 2023',
  icon: <RubyIcon />,
  bgColorIcon: 'orange.500'
};
TransactionRuby.parameters = {
  componentSource: {
    code: transactionRuby
  }
};

export const TransactionGoogle = Template.bind({});
TransactionGoogle.args = {
  label: 'Enrolled',
  content: 'Lorem ipsum dolor sit amet, consectetu',
  score: -3,
  timeDate: '02:15 pm. February 12, 2023',
  icon: <GoogleIcon />,
  bgColorIcon: 'cyan.500'
};
TransactionGoogle.parameters = {
  componentSource: {
    code: transactionGoogle
  }
};
