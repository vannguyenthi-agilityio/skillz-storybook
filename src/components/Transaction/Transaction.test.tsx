// Libraries
import { render, screen } from '@testing-library/react';

// Components
import Transaction from '.';
import { FacebookIcon, RubyIcon } from '@components';

describe('Transaction Component', () => {
  test('should render correctly', () => {
    render(
      <Transaction
        label='Enrolled'
        content='Lorem ipsum dolor sit amet, consectetu'
        score={-3}
        timeDate='02:15 pm. February 12, 2023'
        icon={<FacebookIcon />}
        bgColorIcon='yellow.500'
      />
    );

    const transaction = screen.getByTestId('transaction');
    expect(transaction).toMatchSnapshot();
  });
});

describe('Transaction Component with props icon is Ruby', () => {
  test('should render correctly', () => {
    render(
      <Transaction
        label='Enrolled'
        content='Lorem ipsum dolor sit amet, consectetu'
        score={3}
        timeDate='02:15 pm. February 12, 2023'
        icon={<RubyIcon />}
        bgColorIcon='orange.500'
      />
    );

    const transaction = screen.getByTestId('transaction');
    expect(transaction).toBeTruthy();
  });
});
