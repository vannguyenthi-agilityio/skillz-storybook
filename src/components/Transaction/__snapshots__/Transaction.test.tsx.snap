// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Component should render correctly 1`] = `
<div
  class="css-2yr6ga"
  data-testid="transaction"
>
  <div
    class="css-3yczm5"
  >
    <div
      class="css-1n2mv2k"
    >
      <div
        class="css-1ilnlt6"
      >
        <svg
          fill="none"
          height="34"
          viewBox="0 0 34 34"
          width="34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0.33667 17.0035C0.338609 25.2025 6.30067 32.1835 14.3983 33.4685V21.8202H10.17V17.0035H14.4033V13.3368C14.2141 11.5994 14.8075 9.86781 16.0227 8.61171C17.2379 7.35561 18.9489 6.70518 20.6917 6.83682C21.9426 6.85702 23.1903 6.96843 24.425 7.17016V11.2685H22.3183C21.5931 11.1735 20.8639 11.413 20.3363 11.9196C19.8086 12.4262 19.5396 13.145 19.605 13.8735V17.0035H24.2233L23.485 21.8218H19.605V33.4685C28.3625 32.0845 34.5035 24.0865 33.5792 15.2686C32.6549 6.45077 24.9888 -0.0999629 16.1347 0.362119C7.28061 0.824201 0.338077 8.13734 0.33667 17.0035Z"
            fill="#F3F4F6"
          />
        </svg>
      </div>
      <div
        class="css-14adxt0"
      >
        <p
          class="chakra-text css-13xnkip"
        >
          Enrolled
          :
        </p>
        <p
          class="chakra-text css-in3yi3"
        >
          Lorem ipsum dolor sit amet, consectetu
        </p>
      </div>
    </div>
    <div
      class="css-rhh35x"
    >
      <p
        class="chakra-text css-fwpn6x"
      >
        02:15 pm. February 12, 2023
      </p>
      <div
        class="css-1shx46e"
      >
        <p
          class="chakra-text css-2k2zd"
        >
          -3
        </p>
        <svg
          fill="none"
          height="28"
          viewBox="0 0 28 28"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 24.8715C20.0042 24.8715 24.8715 20.0042 24.8715 14C24.8715 7.99583 20.0042 3.12849 14 3.12849C7.99583 3.12849 3.12849 7.99583 3.12849 14C3.12849 20.0042 7.99583 24.8715 14 24.8715Z"
            fill="black"
          />
          <path
            clip-rule="evenodd"
            d="M28 14C28 21.732 21.732 28 14 28C6.26801 28 0 21.732 0 14C0 6.26801 6.26801 0 14 0C21.732 0 28 6.26801 28 14ZM26.4358 14C26.4358 20.8681 20.8681 26.4358 14 26.4358C7.13192 26.4358 1.56425 20.8681 1.56425 14C1.56425 7.13192 7.13192 1.56425 14 1.56425C20.8681 1.56425 26.4358 7.13192 26.4358 14Z"
            fill="black"
            fill-rule="evenodd"
          />
          <path
            d="M13.324 8.09397C13.5685 7.59848 14.2751 7.59848 14.5196 8.09397L15.8814 10.8532C15.9785 11.05 16.1662 11.1864 16.3834 11.2179L19.4284 11.6604C19.9752 11.7399 20.1935 12.4118 19.7979 12.7975L17.5945 14.9453C17.4373 15.0985 17.3656 15.3191 17.4027 15.5354L17.9229 18.5681C18.0163 19.1127 17.4447 19.528 16.9556 19.2709L14.232 17.839C14.0378 17.7369 13.8058 17.7369 13.6116 17.839L10.888 19.2709C10.3989 19.528 9.82732 19.1127 9.92073 18.5681L10.4409 15.5354C10.478 15.3191 10.4063 15.0985 10.2492 14.9453L8.04575 12.7975C7.65007 12.4118 7.86841 11.7398 8.41523 11.6604L11.4602 11.2179C11.6774 11.1864 11.8651 11.05 11.9622 10.8532L13.324 8.09397Z"
            fill="#FFEBAB"
          />
        </svg>
      </div>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    class="chakra-divider css-iziryy"
  />
</div>
`;
